### PR TITLE
feature(DSG 2576): Employer Invite Users

### PR DESCRIPTION
**Pull request by thechechogomez**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Ticket: https://dol-ewf.atlassian.net/browse/DSG-2576

It allows the employer portal user to invite user by open a dialog and setting an email address and a role. 

**The invite user endpoint is still pending, therefore and invite can't be sent**

## Screenshot of the affected changes

https://github.com/Nuvalence/dsgov-web/assets/60438413/601ca369-a98b-41fb-b71b-34d5d34178a1

## Checklist

- [ ] I have updated the documentation accordingly
- [X] I have confirmed all acceptance criteria defined by the story have been met
- [x] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/employer/feature/dashboard/src/lib/components/account-access/user-list/user-list.component.spec.ts [hunk 1]
```diff
@@ -162,4 +164,17 @@ describe('UserListComponent', () => {
     expect(component.pagingRequestModel.sortBy).toEqual(sortEvent.active);
     expect(component.pagingRequestModel.sortOrder).toEqual(sortEvent.direction.toUpperCase());
   });
+
+  // TODO: Invite user API call
```
Comment (thechechogomez) [inline libs/employer/feature/dashboard/src/lib/components/account-access/user-list/user-list.component.spec.ts:168]: Added this `TODO` comment as the endpoint for inviting users is still pending.

**Review by kevincheng0 — APPROVED** [review]
LGTM

File: libs/employer/feature/dashboard/src/lib/components/account-access/user-action-modal/user-action-modal.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should close the dialog with a valid form', () => {
+    component.formGroup.setValue(formValueMock);
+    const dialog = ngMocks.findInstance(MatDialogRef);
+    const closeSpy = jest.spyOn(dialog, 'close');
+    const submitSPy = jest.spyOn(component, 'submit');
```
Comment (kevincheng0) [inline libs/employer/feature/dashboard/src/lib/components/account-access/user-action-modal/user-action-modal.component.spec.ts:48]: small typo here

Comment (thechechogomez) [reply libs/employer/feature/dashboard/src/lib/components/account-access/user-action-modal/user-action-modal.component.spec.ts:48]: Good catch. I'll fix it on the edit user task as it reuses the same component.

**Pull request merged**